File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        ans = []
        lvl = 0
        for c in s:
            if(c == '('):
                if(lvl>0):
                    ans.append(c)
                lvl+=1
            else:
                lvl-=1
                if(lvl>0):
                    ans.append(c)
        return ''.join(ans)

File: 1078-remove-outermost-parentheses/remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        Stack<Character> stack = new Stack<>();
        StringBuilder sb = new StringBuilder();
        for(char ch : s.toCharArray()){
            if(ch == '('){
                if(!stack.isEmpty()){
                    sb.append(ch);
                }
                stack.push(ch);
            }
            else{
                stack.pop();
                if(!stack.isEmpty()){
                    sb.append(ch);
                }
            }
        }
        return sb.toString();
    }
}